File: 1732. find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int n = gain.length;
        int mx = 0;

        for (int i = 0; i <= n; i++) {
            int alt = 0;
            for (int j = 0; j < i; j++) {
                alt += gain[j];
            }
            mx = Math.max(mx, alt);
        }

        return mx;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.largestAltitude(new int[]{-5, 1, 5, 0, -7})); // Expected: 1
        System.out.println(sol.largestAltitude(new int[]{-4, -3, -2, -1, 4, 3, 2})); // Expected: 0
    }
}

File: 1732. find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain):
        n = len(gain)
        mx = 0

        for i in range(n + 1):
            alt = 0
            for j in range(i):
                alt += gain[j]
            mx = max(mx, alt)

        return mx


if __name__ == "__main__":
    sol = Solution()

    print(sol.largestAltitude([-5, 1, 5, 0, -7]))  # Expected: 1
    print(sol.largestAltitude([-4, -3, -2, -1, 4, 3, 2]))  # Expected: 0